app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+# User
+class User < ApplicationRecord
+  has_many :created_events, foreign_key: 'creator_id', class_name: 'Event'
+  has_many :attendances
+  has_many :event_as_attendee, through: :attendances, source: 'event'
+  validates :name, presence: true, uniqueness: { case_sensitive: false }
+
+  def previous_events
```

Comment:
previous_events and upcoming events in the user model are as same as in events model. these scopes should be built on top of the user model and use user id to retrieve events with that specific user.